docs/proposals/001-proposal-merge-appset-argocd.md

```diff
@@ -0,0 +1,163 @@
+---
+title:  Merging of ApplicationSet codebase into Argo CD
+authors:
+  - "@rishabh625" # Authors' github accounts here.
+sponsors:
+  - TBD        # List all intereste parties here.
+reviewers:
+  - TBD
+  - TBD
+approvers:
+  - TBD
+  - TBD
+
+creation-date: 2022-03-09
+last-updated: 2022-03-10
+---
+
+# Merging of ApplicationSet codebase into Argo CD
+
+This is the proposal to merge codebase of ApplicationSet into Argo CD. ApplicationSet and Argo CD are closely related but yet maintained in different repositories, adding to [issues](https://github.com/argoproj/applicationset/issues/528) wrt releases and support. Creating this proposal to finalize the approach on merging the codebase
+
+## Open Questions [optional]
+
+Before starting need to close open PR's of application set and freeze for no more PR's?
+
+1) merge applicationset v0.4.0 into argo cd
```

Comment:
Based on a conversation in contributors meeting:

* Codefreeze is not required because application set is not getting so many PRs. Instead, we will add a warning to README.md and PR template. The warning should notify contributors that we are in a process of merging and we can ask to hold off very big changes (such as new generator)
* No need to preserve commits history

Can you please mention it in summary?

Reply:
yeah sure thanks!